dask/dataframe/core.py

```diff
@@ -3232,6 +3232,12 @@ def __iter__(self):
             for row in s:
                 yield row
 
+    @derived_from(pd.Series)
+    def __contains__(self, value):
+        for i in range(self.npartitions):
+            s = self.get_partition(i).compute()
+            yield value in s
```

Comment:
Since this returns a generator (which are always truthy) I don't think this is working as expected. For example, this snippet doesn't raise an assertion error:

```python
import dask

ddf = dask.datasets.timeseries()
assert "Alice" in ddf.name
assert "This is not a real name" in ddf.name
```

Reply:
oh sad...

Reply:
Should we even return a concrete value here? Users tend to be caught off guard by these sorts of implicit `compute`s

Reply:
I tried to return a non-computed value, but it seems that python wraps `__contains__` with `__bool__` so you end up running into:

```python-traceback
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-fcc566b58d40> in <module>
----> 1 "Alice" in ddf.name

~/dask/dask/dataframe/core.py in __bool__(self)
    206 
    207     def __bool__(self):
--> 208         raise TypeError(
    209             "Trying to convert {} to a boolean value. Because Dask objects are "
    210             "lazily evaluated, they cannot be converted to a boolean value or used "

TypeError: Trying to convert dd.Scalar<contain..., dtype=int64> to a boolean value. Because Dask objects are lazily evaluated, they cannot be converted to a boolean value or used in boolean conditions like if statements. Try calling .compute() to force computation prior to converting to a boolean value or using in a conditional statement.
```

Given that. I think the clearest behavior would probably be to raise a `NotImplementedError` and recommend that people use `(s.index == key).any()` instead.